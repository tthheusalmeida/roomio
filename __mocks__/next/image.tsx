export default function MockedImage(props: any) {
  return <img {...props} />;
}
