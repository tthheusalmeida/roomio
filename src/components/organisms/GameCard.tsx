import { mergeClassNames } from "@/utils/classNames";
import { useRouter } from "next/navigation";

import Button from "../atoms/Button";

interface GameCardProps {
  title: string;
  image: string;
  online?: number;
  slug: string;
}

export default function GameCard({
  title,
  image,
  online = 0,
  slug,
}: GameCardProps) {
  const router = useRouter();
  const onClickPlay = (path: string) => {
    router.push(`/game/${path}`);
  };

  return (
    <div
      className={mergeClassNames(
        "h-full flex flex-col justify-between",
        "bg-[var(--color-foreground)] border border-violet-200 rounded-lg shadow-sm",
        "bg-violet-900 border-violet-500"
      )}
    >
      <div className="flex flex-col justify-between">
        <img
          className="rounded-t-lg bg-[var(--color-foreground)] block"
          src={image}
          alt={`${title} game preview`}
        />

        <h5 className="p-4 text-lg font-semibold text-violet-900 dark:text-[var(--color-foreground)]">
          {title}
        </h5>
      </div>
      <div className="flex flex-col gap-2 items-start p-4 pt-0">
        {/* <span className="flex gap-2 items-center text-sm">
          <div className="w-3 h-3 border-white border-2 bg-green-500 rounded-full" />
          {online} playing
        </span> */}

        <Button variant="secondary" onClick={() => onClickPlay(slug)}>
          Play
        </Button>
      </div>
    </div>
  );
}
