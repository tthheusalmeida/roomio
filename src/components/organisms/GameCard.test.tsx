import { render, screen, fireEvent } from "@testing-library/react";
import GameCard from "./GameCard";
import { useRouter } from "next/navigation";

jest.mock("next/navigation", () => ({
  useRouter: jest.fn(),
}));

describe("GameCard component", () => {
  const pushMock = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
    (useRouter as jest.Mock).mockReturnValue({ push: pushMock });
  });

  it("should match snapshot", () => {
    const { container } = render(
      <GameCard
        title="Snapshot Game"
        image="/snapshot.jpg"
        slug="snapshot-game"
      />
    );

    expect(container).toMatchSnapshot();
  });

  it("should render title, image and play button", () => {
    render(
      <GameCard
        title="Test Game"
        image="/test.jpg"
        slug="test-game"
        online={5}
      />
    );

    expect(screen.getByText("Test Game")).toBeInTheDocument();
    const image = screen.getByAltText(
      "Test Game game preview"
    ) as HTMLImageElement;
    expect(image).toBeInTheDocument();
    expect(image.src).toContain("/test.jpg");

    const button = screen.getByRole("button", { name: "Play" });
    expect(button).toBeInTheDocument();
  });

  it("should call router.push with correct path when clicking play button", () => {
    render(<GameCard title="Test Game" image="/test.jpg" slug="test-game" />);

    const button = screen.getByRole("button", { name: "Play" });
    fireEvent.click(button);

    expect(pushMock).toHaveBeenCalledWith("/game/test-game");
  });
});
