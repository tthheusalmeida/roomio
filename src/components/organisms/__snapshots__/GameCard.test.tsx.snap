// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`GameCard component should match snapshot 1`] = `
<div>
  <div
    class="h-full flex flex-col justify-between border rounded-lg shadow-sm bg-violet-900 border-violet-500"
  >
    <div
      class="flex flex-col justify-between"
    >
      <img
        alt="Snapshot Game game preview"
        class="rounded-t-lg bg-[var(--color-foreground)] block"
        src="/snapshot.jpg"
      />
      <h5
        class="p-4 text-lg font-semibold text-violet-900 dark:text-[var(--color-foreground)]"
      >
        Snapshot Game
      </h5>
    </div>
    <div
      class="flex flex-col gap-2 items-start p-4 pt-0"
    >
      <button
        class="relative inline-flex justify-center items-center px-4 py-2 gap-4 text-sm font-medium transition-all duration-200 cursor-pointer disabled:cursor-not-allowed rounded-sm w-full bg-violet-100 text-violet-800 hover:enabled:bg-violet-200 disabled:bg-white disabled:text-violet-100"
      >
        <span
          class="flex items-center gap-4 visible"
        >
          Play
        </span>
      </button>
    </div>
  </div>
</div>
`;
